opm/parser/eclipse/EclipseState/Tables/TableManager.cpp

```diff
@@ -337,8 +333,7 @@ namespace Opm {
         auto& container = forceGetTables(keywordName , numTables);
         for (size_t tableIdx = 0; tableIdx < keyword->size(); ++tableIdx) {
             const auto tableRecord = keyword->getRecord( tableIdx );
-            std::shared_ptr<PlyrockTable> table = std::make_shared<PlyrockTable>();
-            table->init( tableRecord );
+            std::shared_ptr<PlyrockTable> table = std::make_shared<PlyrockTable>(tableRecord);
```

Comment:
for the record: I consider making these constructors publicly available dangerous because whole tables can be defaulted and the code which instantiates these objects should be aware of this. If you have a different opinion, I have no issues with public constructors, too.